app/lib/oidc_client.rb

```diff
@@ -53,12 +55,14 @@ def tokens!(oidc_nonce: nil)
   end
 
   def userinfo(access_token:, refresh_token:)
-    response = oauth_request(
-      access_token: access_token,
-      refresh_token: refresh_token,
-      method: :get,
-      uri: userinfo_endpoint,
-    )
+    response = govuk_timer "userinfo" do
+      oauth_request(
```

Comment:
Nit: I found the nested block somewhat difficult to read as it's not obvious the govuk_timer just returns the result of the block. Not sure if it's possible to set the attribute within the block instead?

```suggestion
    govuk_timer "userinfo" do
      response = oauth_request(
```

(This shouldn't block this PR)

Reply:
How about if I changed the method name to `time_and_return`?

Reply:
Yup, that'd be improvement!